### PR TITLE
fix(ses-actions): permissions too wide for S3 action

### LINKED ISSUE #29811: aws_ses: SES ReceiptRuleSet S3 action grants too wide permissions
### Describe the bug

SES ReceiptRuleSet S3 action grants too wide permissions

### Expected Behavior

Should work as documented
https://docs.aws.amazon.com/ses/latest/dg/receiving-email-permissions.html#receiving-email-permissions-s3

Currently missing the following block:

```
      "Condition":{
        "StringEquals":{
          "AWS:SourceAccount":"111122223333",
          "AWS:SourceArn": "arn:aws:ses:region:111122223333:receipt-rule-set/rule_set_name:receipt-rule/receipt_rule_name"
        }
      }

```

### Current Behavior

https://github.com/aws/aws-cdk/blob/a7384c282756890a3e211c064b4e8a2dee3dab2a/packages/aws-cdk-lib/aws-ses-actions/test/actions.test.ts#L186

### Reproduction Steps

```ts
    const ruleSet = new ses.ReceiptRuleSet(this, 'RuleSet');

    const defaultRule = ruleSet.addRule('DefaultRule', {
      recipients: props.recipients,
      enabled: true,
    });

    defaultRule.addAction(new actions.S3({
      bucket: new s3.Bucket(this, 'EmailBucket', {
        removalPolicy: cdk.RemovalPolicy.DESTROY,
        bucketName: props.bucketName,
      }),
    }));

```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### CDK CLI Version

2.137.0

### Framework Version

_No response_

### Node.js Version

20

### OS

all

### Language

TypeScript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by msambol**
This matches the condition described [here](https://docs.aws.amazon.com/ses/latest/dg/receiving-email-permissions.html#receiving-email-permissions-s3).

Closes #29811.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by msambol** [comment]
Exemption request: Per https://github.com/aws/aws-cdk/blob/f10494cab79c033d7913a435b0e2db12fe59698d/packages/%40aws-cdk-testing/framework-integ/test/aws-ses-actions/test/integ.actions.ts#L10-L18, this integration test cannot be deployed.

File: packages/aws-cdk-lib/aws-ses-actions/lib/s3.ts [hunk 1]
```diff
@@ -55,7 +55,8 @@ export class S3 implements ses.IReceiptRuleAction {
       resources: [this.props.bucket.arnForObjects(`${keyPattern}*`)],
       conditions: {
         StringEquals: {
-          'aws:Referer': cdk.Aws.ACCOUNT_ID,
+          'aws:SourceAccount': cdk.Aws.ACCOUNT_ID,
+          'aws:SourceArn': `arn:${cdk.Aws.PARTITION}:ses:${cdk.Aws.REGION}:${cdk.Aws.ACCOUNT_ID}:receipt-rule-set/*:receipt-rule/*`,
```
Comment (msambol) [inline packages/aws-cdk-lib/aws-ses-actions/lib/s3.ts:59]: I think this is the closest we can get, given we don't have `rule_set_name` or `receipt_rule_name`.

**Comment by msambol** [comment]
I think someone on the CDK team will need to run the integration test. Wherever `cdk-ses-receipt-test@yopmail.com` is verified.

File: packages/aws-cdk-lib/aws-ses-actions/lib/s3.ts (hunk 1 above)
Comment (nmussy) [reply packages/aws-cdk-lib/aws-ses-actions/lib/s3.ts:59]: You have access to `rule.receiptRuleName`, is it not "the name of the receipt rule that contains the deliver to Amazon S3 bucket action"?

Comment (nmussy) [reply packages/aws-cdk-lib/aws-ses-actions/lib/s3.ts:59]: Correct me if I'm wrong, but we're not getting a lot of additional restraints on the allocated perms with the `SourceArn` condition in its current state. The only thing we're gaining is the region restriction. The uncoditional `receipt-rule-set` doesn't help much, given we already only had the SES principal. I think trying to retrieve both the `ruleSet` from the `ReceiptRuleProps`, and the `receiptRuleName` (either generated, passed as a constructor prop, or imported), should be a priority

Comment (msambol) [reply packages/aws-cdk-lib/aws-ses-actions/lib/s3.ts:59]: Yeah, perhaps why this `Condition` is the way it is. May close this PR, I think `aws:Referer` with account ID is enough.

**Comment by msambol** [comment]
Closing in favor of https://github.com/aws/aws-cdk/pull/29833. Thanks, @nmussy !

**Pull request closed without merging**